.pre-commit-config.yaml

```
@@ -0,0 +1,15 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.6.0
+  hooks:
+    - id: check-added-large-files
+    - id: check-yaml
+    - id: end-of-file-fixer
+    - id: trailing-whitespace
```

Can also add these hooks from https://github.com/Lucas-C/pre-commit-hooks that handle LF file endings and 644 permissions?

```suggestion
    - id: trailing-whitespace
- repo: https://github.com/Lucas-C/pre-commit-hooks
  rev: v1.5.5
  hooks:
    - id: forbid-crlf
    - id: remove-crlf
    - id: chmod
      args: ['644']
```

---

Looks good.

---

Hmm, not sure why, but pre-commit.ci seeems to think our files have 664 permission instead of 644?

---

I've added back the GitHub Action workflows for 644 permission checks/formatting for now at commit 5a01ad61b354a832fd956deb8da90b22ac1e22e1.